### ISSUE TITLE
explore different mutation observer model

### ISSUE THREAD

**Issue by SilentCicero** [SilentCicero/throw-down#1]
Right now we are using one central mutation observer to watch the document.body. This works fairly well, however, what if we could create a faster model where mutation observers are created for each component. Not sure what would be faster. But so far, just observing the body seems to be fairly quick.

https://github.com/skatejs/skatejs/blob/5486aeb802c9d6350355acdc00a8fb469634a5d0/lib/shared/document-observer.js

**Comment by yoshuawuyts** [comment]
Yeah it would be heaps rad if mutation observers could be created on a per-component basis; it would mean that widgets can be created that can be shared between projects, without needing to plug in a custom observe method each time

**Comment by SilentCicero** [comment]
@yoshuawuyts late reply, the code is simple enough to modify, if you have time/pr I would totally accept it  =D

**Comment by yoshuawuyts** [comment]
haha, it ok - so [on-load](https://github.com/shama/on-load) kind of does what we want it to do now C:
